akka-stream/src/main/java/akka/stream/javadsl/AsPublisher.java

```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.javadsl;
+
+public enum AsPublisher {
+  WITH_FANOUT, WITHOUT_FANOUT
+}
```

Comment:
Perhaps not an issue, but was thinking to keep the same style for those in all places.
e.g. the new Http APIs proposal enum is not enough (as needs to be configurable), so I was thinking if exposing the enum directly is a good idea or not (could be an enum backing 2 impls of a interface, but we get those via methods).